src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs

```diff
@@ -670,8 +670,7 @@ internal bool TryGetValue(int index, out DateTime value)
 
             Debug.Assert(segment.IndexOf(JsonConstants.BackSlash) == -1);
 
-            if (segment.Length <= JsonConstants.MaximumDateTimeOffsetParseLength
```

Comment:
Given this check is redundant, I am assuming that test coverage would show/confirm this condition was never false. I will verify this tomorrow by running code coverage (against what's in master).

Reply:
> I am assuming that test coverage would show/confirm this condition was never false.

Yes, AFAICS, because of the `IsValidDateTimeOffsetParseLength` check above:

https://github.com/dotnet/corefx/blob/bd18c12df9e5dd365a8b708cc2e7e0488b8177c1/src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs#L659-L663

I think this check should always be `true`.

> I will verify this tomorrow by running code coverage (against what's in master).

That would be great, thanks! 🙏

Reply:
Hmm. The only exception I can think of would be if the segment is escaped at this point, but all of the calls have

```csharp
Debug.Assert(segment.IndexOf(JsonConstants.BackSlash) == -1);
```

Right above it.

Reply:
> I will verify this tomorrow by running code coverage (against what's in master).

Yep, the branches are never false.
JsonDocument:
![image](https://user-images.githubusercontent.com/6527137/67543109-a08a5300-f6a4-11e9-8f65-426dff4e0583.png)

Utf8JsonReader:
![image](https://user-images.githubusercontent.com/6527137/67543180-024abd00-f6a5-11e9-9578-147b2bce8853.png)

Reply:
Nice. What do I do to get the remaining CI checks green? 🤔